File: Poin A/Memory.java
public class Memory extends Hardware 
{
    private String frequency;
    private String memorySize;
    private String supportsCuda;

    public Memory(String frequency, String memorySize, String supportsCuda) {
        this.frequency = frequency;
        this.memorySize = memorySize;
        this.supportsCuda = supportsCuda;
    }

    public Memory() {
    }

    public String getSupportsCuda() {
        return supportsCuda;
    }
    public String getFrequency() {
        return frequency;
    }
   
    public String getMemorySize() {
        return memorySize;
    }

    public void setFrequency(String frequency) {
        this.frequency = frequency;
    }
    public void setMemorySize(String memorySize) {
        this.memorySize = memorySize;
    }
    public void setSupportsCuda(String supportsCuda) {
        this.supportsCuda = supportsCuda;
    }

    public void output_memory(){
        System.out.println("Id Product          : "+ this.getId_Product());
        System.out.println("Price               : "+ this.getPrice());
        System.out.println("Brand Hardware      : "+ this.getBrand());
        System.out.println("Model Hardware      : "+ this.getModel());
        System.out.println("Frequency Memory    : "+ this.getFrequency());
        System.out.println("Memory Size         : "+ this.getMemorySize());
        System.out.println("Supports Cuda       : "+ this.getSupportsCuda());
    }
    
    
}

File: Poin B.cpp/Memory.cpp
#include<string>
#include<iostream>
#include<string>
//Include file parents
#include"Hardware.cpp"

using namespace std;


class Memory : public Hardware{
    private :
        //Deklarasi Atribut
        string frequency;
        string memorySize;
        string supportCuda;
        
    public :
        //Constructor
        Memory(){

        }
        
        //Get Set Methods
        void setFrequency(string frequency){
            this->frequency = frequency;
        }

        string getFrequency(){
            return this->frequency;
        }

        void setMemorySize(string memorySize){
            this->memorySize = memorySize;
        }

        string getMemorySize(){
            return this->memorySize;
        }

        void setSupportCude(string supportCuda){
            this->supportCuda = supportCuda;
        }

        string getSupportCuda(){
            return this->supportCuda;
        }

        //Output Methods
        void output_Memory(){
            cout << "Id Product                 : " << this->getId_Product() << endl;
            cout << "Price                      : " << this->getPrice() << endl;
            cout << "Brand                      : " << this->getBrand() << endl;
            cout << "Model                      : " << this->getModel() << endl;
            cout << "Frequency                  : " << this->getFrequency() << endl;
            cout << "Memory Size                : " << this->getMemorySize() << endl;
            cout << "Support Cuda               : " << this->getSupportCuda() << endl;
        }
        
    
        //Destructor
        ~Memory(){

        }

        
};
